src/structure/4C_structure_timint_impl.cpp

```diff
@@ -3171,6 +3171,42 @@ void Solid::TimIntImpl::cmt_linear_solve()
       cmtbridge_->get_strategy().build_saddle_point_system(
           stiff_, fres_, disi_, dbcmaps_, blockMat, blocksol, blockrhs);
 
+      // compute the nullspace vectors for the Lagrange multiplier field for MueLu
+      if (contactsolver_->params().isSublist("Belos Parameters"))
+        if (contactsolver_->params()
+                .sublist("Belos Parameters")
+                .get<std::string>("Preconditioner Type") == "ML")
+        {
+          int dim_nullspace = contactsolver_->params()
+                                  .sublist("Inverse2")
+                                  .sublist("MueLu Parameters")
+                                  .get<int>("PDE equations", -1);
+
+          // get the degree of freedom map from the block matrix
+          Epetra_Operator* raw_block_mat = blockMat.get();
+          std::shared_ptr<Core::LinAlg::BlockSparseMatrixBase> block_mat_blocked_operator =
+              std::dynamic_pointer_cast<Core::LinAlg::BlockSparseMatrixBase>(
+                  std::shared_ptr<Epetra_Operator>(raw_block_mat, [](Epetra_Operator*) {}));
+          auto mat11 = block_mat_blocked_operator->matrix(1, 1);
+          std::shared_ptr<Epetra_Map> nullspace_map =
+              std::make_shared<Epetra_Map>(mat11.range_map());
+          const Epetra_Map& dofmap = *nullspace_map;
```

Comment:
1. I think, this can be shortened.
2. Why range map and not row map?
```suggestion
          const Epetra_Map& dofmap = mat11.row_map();
```